src/experimental/slam_toolbox_lifelong_node.cpp

```diff
@@ -29,6 +29,9 @@ int main(int argc, char ** argv)
     auto temp_node = std::make_shared<rclcpp::Node>("slam_toolbox");
     temp_node->declare_parameter("stack_size_to_use");
     if (temp_node->get_parameter("stack_size_to_use", stack_size)) {
+#ifdef _WIN32
+      RCLCPP_WARN(temp_node->get_logger(), "Can't dynamically change stack size on Windows. Node using stack size 40000000");
```

Comment:
Is there another API to change the stack size then? This is kind of an important element for serialization / deserialization of large maps

Reply:
From my research, I wasn't able to find a way to dynamically change the stack size of the program at runtime on Windows. However, I was able to find a way to change the stack size when a thread is created [here](https://docs.microsoft.com/en-us/windows/win32/procthread/thread-stack-size).

Do you think it would be possible to utilize this functionality instead? Or, instead of serializing and deserializing maps on the stack, is it possible to use dynamic memory, just for Windows?

Reply:
It uses boost serialization, so that's totally out of our control where that memory is located for serialization. 

So are you suggesting we serialize/deserialize in another thread that in instantiation we set the max size on? That would need some testing on your end to validate, but I could approve that.

Reply:
Instead of putting only the serialization/deserialization on a separate thread, it might be easier to just put the entire node in a separate thread. Do you think that would be fine too?

Reply:
I would not like to do that, that would add an extra thread for everyone for all of run-time, versus only an extra thread on serialization/deserialization processes

Reply:
I see, that makes sense! Would I just have to modify `SlamToolbox::serializePoseGraphCallback` and `SlamToolbox::loadSerializedPoseGraph`? Are there any other functions that do the serialization and deserialization?

Reply:
Yup, that's right

Reply:
None that I can share publicly